routes/errortracker.js

```diff
@@ -0,0 +1,257 @@
+    filteredMessageOrException.forEach(function filter(msg){
+        if(errorEvent.Message.toString().includes(msg) || errorEvent.Exception.toString().includes(msg)){
+            return true;
+        }
+    });
+    return false;
+
+}
+
+/**
+ * Function that handles errors that come from an attempt to write to the logs
+ * @param err error
+ * @param res http response object
+ */
+function logWritingError(err,res) {
+    if(err) {
+        res.status(statusCodes.INTERNAL_SERVER_ERROR).send({error:'Cannot write to Google Cloud Logging'});
+        winston.error(projectId, 'Cannot write to Google Cloud Logging: '+params['v'],err)
+    }
+
+
+}
+
+/**
+ *extract params in GET request from query and fill errorEvent object
+ * @param req
+ * @param res
+ * @param next
+ */
+function getHandler(req,res,next){
+
+    params = url.parse(req.url, true).query;
+    var referer = req.get('Referer').toString();
+    var resource = {
+        type:'compute.googleapis.com',
+        labels : {
+            "compute.googleapis.com/resource_type": "logger",
+            "compute.googleapis.com/resource_id":   "errors"
+        }
+    };
+    var line = params['l'];
+    var errorType = 'default';
+    var isUserError = false;
+    if(params['a'] === 1){
+        errorType = 'assert';
+        isUserError = true;
+    }
+    //log as INFO severity by default, because reports are spammy
+
+    var severity = 'INFO';
+    var level = errorLevels.Info;
+    // if request comes from the cache and thus only from valid AMP docs we log as "Error"
+    var isCdn =  false;
+    if(referer.startsWith('https://cdn.ampproject.org/') ||
+    referer.includes('cdn.ampproject.org/') ||
+    referer.includes('.ampproject.net/')) {
+        severity = 'ERROR';
+        level = errorLevels.Error;
+        errorType += '-cdn';
+        isCdn = true;
+    } else {
+        errorType += '-origin'
+    }
+    var is3p = false;
+    var runTime = params['rt'];
+    if (runTime !== '') {
+        errorType +=  '-' + runTime;
+        if(runTime === 'inabox'){
+            severity = 'ERROR';
+            level = errorLevels.Error;
+        }
+        if(runTime === '3p'){
+            is3p = true;
+
+        }
+    } else{
+        if(params['3p'] === 1){
+            is3p = true;
+            errorType += '-3p';
+        } else {
+            errorType += '-1p';
+        }
+    }
+    var isCanary = false;
+    if(params['ca'] === 1){
+        errorType += '-canary';
+        isCanary = true;
+    }
+    if(params['ex'] === 1){
+        errorType += '-expected';
+    }
+    var sample = Math.random(); 
+    var throttleRate = 0.01;
+    
+    if(isCanary){
+        throttleRate = 1.0 //explicitly log all errors
+
+    } else if(is3p) {
+        throttleRate = 0.1;
+
+    } else if(isCdn){
+        throttleRate = 0.1;
+    }
+
+    if(isUserError) {
+        throttleRate = throttleRate/10;
+    }
+
+    if (!sample <= throttleRate) {
```

Comment:
This `!` is incorrect. We need to flip the test itself: `if (sample > throttleRate)`